fastly/custom_tls_activation_test.go

```diff
@@ -0,0 +1,165 @@
+package fastly
+
+import (
+	"testing"
+)
+
+func TestClient_TLSActivation(t *testing.T) {
+	t.Parallel()
+
+	fixtureBase := "custom_tls_activation/"
+
+	// Create
+	var err error
+	var ta *TLSActivation
+	record(t, fixtureBase+"create", func(c *Client) {
+		ta, err = c.CreateTLSActivation(&CreateTLSActivationInput{
+			TLSCertificate:   &TLSCertificate{ID: "CERTIFICATE_ID"},
+			TLSConfiguration: &TLSConfiguration{ID: "CONFIGURATION_ID"},
+			TLSDomain:        &TLSDomain{ID: "DOMAIN_NAME"},
+		})
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Ensure deleted
+	defer func() {
+		record(t, fixtureBase+"cleanup", func(c *Client) {
+			c.DeleteTLSActivation(&DeleteTLSActivationInput{
+				ID: ta.ID,
+			})
+		})
+	}()
+
+	// List
+	var lta []*TLSActivation
+	record(t, fixtureBase+"list", func(c *Client) {
+		lta, err = c.ListTLSActivations(&ListTLSActivationsInput{})
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(lta) < 1 {
+		t.Errorf("bad TLS activations: %v", lta)
+	}
+	if lta[0].TLSCertificate == nil {
+		t.Errorf("TLS certificate relation should not be nil: %v", lta)
+	}
+	if lta[0].TLSConfiguration == nil {
+		t.Errorf("TLS configuration relation should not be nil: %v", lta)
+	}
+	if lta[0].TLSDomain == nil {
+		t.Errorf("TLS domain relation should not be nil: %v", lta)
+	}
+
+	// Get
+	var gta *TLSActivation
+	record(t, fixtureBase+"get", func(c *Client) {
+		gta, err = c.GetTLSActivation(&GetTLSActivationInput{
+			ID: ta.ID,
+		})
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	if ta.ID != gta.ID {
+		t.Errorf("bad ID: %q (%q)", ta.ID, gta.ID)
+	}
+
+	// Update
+	var uta *TLSActivation
+	record(t, fixtureBase+"update", func(c *Client) {
+		uta, err = c.UpdateTLSActivation(&UpdateTLSActivationInput{
+			ID:             "ACTIVATION_ID",
+			TLSCertificate: &TLSCertificate{},
+		})
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	if ta.ID != uta.ID {
+		t.Errorf("bad ID: %q (%q)", ta.ID, uta.ID)
+	}
+
+	// Delete
+	record(t, fixtureBase+"delete", func(c *Client) {
+		err = c.DeleteTLSActivation(&DeleteTLSActivationInput{
+			ID: ta.ID,
+		})
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestClient_CreateTLSActivation_validation(t *testing.T) {
```

Comment:
This is not actually testing the validation business logic, only the happy path which we already have covered. It's more common for these functions to assert the error handling business logic of the methods, such as asserting that we returned a specific error message if an empty field was passed etc. Here is an example of what I mean: https://github.com/fastly/go-fastly/blob/master/fastly/datadog_test.go#L166-L182 

This applies generally to the other of validation tests elsewhere in this PR.

Reply:
That makes sense. I will add these. I was following platform_tls_test.go and it does not have these "not happy" paths tested.